archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterGenericMethodReturnTypesTest.java

```diff
@@ -0,0 +1,734 @@
+        JavaType genericReturnType = new ClassFileImporter().importClasses(SomeClass.class, ClassParameterWithSingleTypeParameter.class, String.class, File.class)
+                .get(SomeClass.class).getMethod("method").getReturnType();
+
+        assertThatType(genericReturnType).as("generic return type").hasActualTypeArguments(
+                parameterizedType(ClassParameterWithSingleTypeParameter.class)
+                        .withWildcardTypeParameterWithUpperBound(String.class),
+                parameterizedType(ClassParameterWithSingleTypeParameter.class)
+                        .withWildcardTypeParameterWithLowerBound(File.class)
+        );
+    }
+
+    @Test
+    public void imports_generic_method_return_type_with_actual_type_arguments_with_multiple_wildcards_with_various_bounds() {
+        @SuppressWarnings("unused")
+        class GenericReturnType<A, B> {
+        }
+        @SuppressWarnings("unused")
+        class SomeClass {
+            GenericReturnType<
+                    ClassParameterWithSingleTypeParameter<Map<? extends Serializable, ? super File>>,
+                    ClassParameterWithSingleTypeParameter<Reference<? super String>>> method() {
+                return null;
+            }
+        }
+
+        JavaType genericReturnType = new ClassFileImporter()
+                .importClasses(
+                        SomeClass.class, ClassParameterWithSingleTypeParameter.class,
+                        Map.class, Serializable.class, File.class, Reference.class, String.class)
+                .get(SomeClass.class).getMethod("method").getReturnType();
+
+        assertThatType(genericReturnType).as("generic return type").hasActualTypeArguments(
+                parameterizedType(ClassParameterWithSingleTypeParameter.class)
+                        .withTypeArguments(parameterizedType(Map.class)
+                                .withWildcardTypeParameters(
+                                        wildcardType().withUpperBound(Serializable.class),
+                                        wildcardType().withLowerBound(File.class))),
+                parameterizedType(ClassParameterWithSingleTypeParameter.class)
+                        .withTypeArguments(parameterizedType(Reference.class)
+                                .withWildcardTypeParameterWithLowerBound(String.class))
+        );
+    }
+
+    @Test
+    public void imports_type_variable_as_generic_method_return_type() {
+        @SuppressWarnings("unused")
+        class SomeClass<T extends String> {
+            T method() {
+                return null;
+            }
+        }
+
+        JavaType genericMethodReturnType = new ClassFileImporter().importClasses(SomeClass.class, String.class)
+                .get(SomeClass.class).getMethod("method").getReturnType();
+
+        assertThatType(genericMethodReturnType).as("generic method return type")
+                .isInstanceOf(JavaTypeVariable.class)
+                .hasErasure(String.class);
+    }
```

Comment:
Yeah, you're right :disappointed: I had some kind of tunnel vision ignoring the method type definition. This is a whole new category of tests, since we can have all kinds of generic method type parameter definitions :see_no_evil: